### DISCUSSION TITLE
How to get Sentry to log errors within try-catch

### DISCUSSION THREAD

**Discussion by Jedeu** [nuxt-community/sentry-module#428] [Q&A]
Hi I'm seeing Sentry send uncaught errors but nothing found within a try-catch block. I read [this](https://github.com/nuxt-community/sentry-module/issues/406) earlier question and wanted to see if my setup might be preventing Sentry from sending the errors over. Thanks in advance!

**server/index.js (entrypoint)**
```
const Routers = require('./routers');
const config = require('./config');
app.use('baseUrl', Routers);
const nuxt = new Nuxt(config);
await nuxt.ready();

app.use(nuxt.render);
```

**routers.js**
```
  const router = require('express').Router();
  const MySpace = require('./services/MySpace');

  router.use('/my_space', MySpace);
```

**MySpace.js**

```
  async doTheThing(req, res) {
    try {
      // execute some logic
    } catch (e) {
      // process.sentry.captureException(e) <--- I need to manually add this if I want to send stuff to Sentry, hoping to send any errors over automatically
      return res.status(500).json(e);
    }
  }

  Router.post('/do_the_thing', doTheThing);
```

**nuxt.config.js**
```
sentry: {
    dsn: myDsn,
    tracing: true,
    config: {
      debug: true,
    },
},
```

**Comment by rchl** [comment]
Caught exceptions can't be captured automatically. That's the whole point of catching them. You need to manually report them using `process.sentry.captureException(e)` as you showed.

Alternatively you can redesign your app so that the route handlers don't catch anything itself and then you add a global error handler as a last route that captures the exception and sends a 500 response. Then you would only need to capture in one place.

Or if you don't want control over what is sent in the response then just don't try/catch exceptions and don't add an error handler. Then Sentry will capture those automatically and your app will respond with a generic 500 error.

**Reply by rchl** [reply]
Also note that express.js doesn't have built-in support for handling async routes so you'd need to create a wrapper yourself that detects when something throws and passes the error to the `next` handler. Something like this:

```js
export function asyncMiddleware(endpointHandlerFunction) {
    return function(req, res, next) {
        Promise.resolve(endpointHandlerFunction(req, res, next)).catch(next);
    };
}

// and then

Router.post('/do_the_thing', asyncMiddleware(doTheThing));
```

**Reply by Jedeu** [reply]
Ah perfect, thanks for the info!
